File: ddd/tests/fixtures/operation-error-set/typescript-class-workspace/billing-domain/src/missing.ts
/** Each operation leaves out one case the mapping names. */

import type { Result } from "./result.ts";

export type IssueInvoiceError = "already-issued";
export type OpenInvoiceError = "negative-amount";

export class Invoice {
  static open(): Result<Invoice, OpenInvoiceError> {
    return { ok: true, value: new Invoice() };
  }

  issue(): Result<void, IssueInvoiceError> {
    return { ok: true, value: undefined };
  }
}

File: ddd/tests/fixtures/operation-error-set/rust-workspace/billing-domain/src/missing.rs
//! Each operation leaves out one case the mapping names.

pub enum IssueInvoiceError {
    AlreadyIssued,
}

pub enum OpenInvoiceError {
    NegativeAmount,
}

pub struct Invoice;

impl Invoice {
    pub fn issue(&mut self) -> Result<(), IssueInvoiceError> {
        Ok(())
    }

    pub fn open() -> Result<Self, OpenInvoiceError> {
        Ok(Self)
    }
}
